pkg/generators/enum.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/types"
+)
+
+const tagEnumType = "enum"
+
+type enumValue struct {
+	Name    string
+	Value   string
+	Comment string
+}
+
+// enumMap is a map of enum type name to all its values
+type enumMap map[types.Name][]*enumValue
+
+type enumContext struct {
+	enumTypes enumMap
+}
+
+func newEnumContext(c *generator.Context) *enumContext {
+	return &enumContext{enumTypes: parseEnums(c)}
+}
+
+// IsKnownEnumType checks if a given type is a known enum type
+func (ec *enumContext) IsKnownEnumType(t *types.Type) bool {
+	_, ok := ec.enumTypes[t.Name]
+	return ok
+}
+
+// FormatEnumValues returns all possible values of the enum type
+//  the results are sorted and quoted as Go literals.
```

Comment:
Nit: A lot of your comments have a weird structure/indentation/spaces inserted.

Reply:
Misconfiguration of the editor (with my own workstation). Fixed.